examples/configure_pin_function.rs

```diff
@@ -0,0 +1,41 @@
+//! Configure pin 6 with a pull-down resistor, and sample it to control
+//! whether the LED is flashing or not.
+
+#![no_std]
+#![no_main]
+
+use teensy4_panic as _;
+
+use cortex_m_rt::entry;
+use embedded_hal::digital::v2::OutputPin;
+use imxrt_hal::{
+    gpio::{Input, GPIO},
+    iomuxc::{configure, gpio::Pin, Config, Hysteresis, PullKeeper, IOMUX},
+};
+use teensy4_bsp::{configure_led, t40, Peripherals};
+
+fn configure_switch<P: Pin + IOMUX>(mut switch_pin: P) -> GPIO<P, Input> {
```

Comment:
Ah! that must be because I initially was trying to pass a reference to switch_pin.

I guess passing a reference to a pin doesn't have any benefits, but I was confused because in one of my incantations, rustc was telling me that IOMUX wasn't implemented for &P.

Reply:
I don't recall the exact error. I thought that adding IOMUX as a trait was one of the things that fixed it.

Reply:
Yeah, I tried again to get configure_switch working with a `&mut P` and I had no luck. 

Given what I've learned in the past day, I'm not sure that it could ever work because pins can't be moved out of `pins` index to prevent duplication.

Reply:
Sorry I missed all these replies! Thanks for sharing your research.

With `configure_switch` as written, a `&mut P` won't cut it, because [`GPIO::new` takes ownership of the pad](https://docs.rs/imxrt-hal/0.4.5/imxrt_hal/gpio/struct.GPIO.html#method.new). In order to pass ownership to the constructor, `configure_switch` must also take ownership.

If we separate pin configuration from GPIO construction, like

```rust
fn configure_switch<P: Pin>(switch_pin: &mut P) {
    const LOW_SWITCH_CONFIG: Config = Config::zero()
        .set_hysteresis(Hysteresis::Enabled)
        .set_pull_keeper(Some(PullKeeper::Pulldown100k));
    configure(switch_pin, LOW_SWITCH_CONFIG);
    // configure(&mut switch_pin, LOW_SWITCH_CONFIG);
}

// ...
let switch = {
    configure_switch(&mut pins.p6);
    GPIO::new(pins.p6)
};
```

we can still get away without the `IOMUX` constraint.

Notice the commented out `configure()` call. If we try to compile that line, the compiler throws the "IOMUX not implemented" error. I think that's because we're trying to pass a `&mut &mut P` to a function that expects `&mut P`. I'm actually not sure if there's a way to make that form work, but now I'm curious...